pkg/network/ebpf/c/protocols/kafka/kafka-parsing.h

```diff
@@ -1804,4 +1808,22 @@ static __always_inline void update_topic_name_size_telemetry(kafka_telemetry_t *
     __sync_fetch_and_add(&kafka_tel->topic_name_size_buckets[bucket_idx], 1);
 }
 
+// update_classified_api_version_hits_telemetry updates arrays keeping track of api versions hits in the telemetry
+static __always_inline void update_classified_api_version_hits_telemetry(kafka_telemetry_t *kafka_tel, s16 api_key, s16 api_version) {
+    __u8 bucket_idx = api_version - 1;
+
+    // Ensure that the bucket index falls within the valid range.
+    bucket_idx = bucket_idx < 0 ? 0 : bucket_idx;
+    bucket_idx = bucket_idx > (KAFKA_TELEMETRY_API_VERSIONS_NUM_OF_BUCKETS - 1) ? (KAFKA_TELEMETRY_API_VERSIONS_NUM_OF_BUCKETS - 1) : bucket_idx;
+
+    switch (api_key) {
+    case KAFKA_FETCH:
+        __sync_fetch_and_add(&kafka_tel->classified_fetch_api_version_hits[bucket_idx], 1);
+        break;
+    case KAFKA_PRODUCE:
+        __sync_fetch_and_add(&kafka_tel->classified_produce_api_version_hits[bucket_idx], 1);
+        break;
+    }
+}
```

Comment:
you took the implementation from the topic_name telemetry, I think that's incorrect, or at least, needs to be modified per-fetch and produce to adjust the number of buckets to the correct size

Reply:
Not sure I got you, it's not really buckets it's just counters where index = api_verison so I'm just doing +1 to the correct counter (maybe I should rename bucket -> counter?)
But I changed the implementation to use index 0 for unknowns

Reply:
i think you should split it per api key, as the shared const, makes it a little bit messy